### PR TITLE
AEA-1230 protolint check

### PR DESCRIPTION
**Pull request by solarw**
## Proposed changes

protolint check

## Fixes

https://github.com/fetchai/agents-aea/issues/2136

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: protolint.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+lint:
+  rules:
+    remove:
+      - MESSAGE_NAMES_UPPER_CAMEL_CASE
+      - ENUM_FIELD_NAMES_ZERO_VALUE_END_WITH
+      - PACKAGE_NAME_LOWER_CASE
+      - REPEATED_FIELD_NAMES_PLURALIZED
+      - FIELD_NAMES_LOWER_SNAKE_CASE
```
Comment (DavidMinarsch) [inline protolint.yaml:8]: We should try to fix `MESSAGE_NAMES_UPPER_CAMEL_CASE`, `REPEATED_FIELD_NAMES_PLURALIZED` and `FIELD_NAMES_LOWER_SNAKE_CASE`.

File: .github/workflows/workflow.yml [hunk 2]
```diff
@@ -152,8 +153,27 @@ jobs:
...
+          go-version: '^1.14.0'
+      - name: Install protolint (ubuntu-latest)
+        run: |
+          sudo apt-get update --fix-missing
+          sudo apt-get autoremove
+          sudo apt-get autoclean
+          make protolint_install
+      - name: Protolint check
+        run: |
+          make protolint
```
Comment (marcofavorito) [inline .github/workflows/workflow.yml:176]: What do you think in adding a testing environment in `tox.ini`? So we can call `tox -e protolint` and have uniformity with other test environments.

File: aea/protocols/generator/base.py [hunk 3]
```diff
@@ -1921,13 +1923,32 @@ def generate_protobuf_only_mode(self) -> None:
...
+        if sys.platform.startswith("win"):
+            protolint_base_cmd = "protolint"
+        else:
+            protolint_base_cmd = "PATH=${PATH}:${GOPATH}/bin/:~/go/bin protolint"
+
+        if call(f"{protolint_base_cmd} version", shell=True) != 0:  # nosec
+            raise ValueError("protolint is not installed!")
+
+        cmd = f'{protolint_base_cmd} lint -fix "{filepath}"'
+        call(cmd, shell=True)  # nosec
```
Comment (marcofavorito) [inline aea/protocols/generator/base.py:1950]: `shell=True` is not recommended for security reason, yet I think we have good control on the `cmd` variable. Maybe some checks on `filepath`? E.g. check that it is actually a path.

File: aea/protocols/generator/common.py [hunk 4]
```diff
@@ -364,7 +364,7 @@ def _create_protocol_file(
     """
     pathname = os.path.join(path_to_protocol_package, file_name)
 
-    with open(pathname, "w") as file:
+    with open(pathname, "w", newline="\n") as file:
```
Comment (marcofavorito) [inline aea/protocols/generator/common.py:367]: why this? I guess related to #1631

**Review by marcofavorito — APPROVED** [review]
LGTM. I wouldn't mind to see a new line in `scripts/generate_all_protocols.py` to check `protolint` is installed:

https://github.com/fetchai/agents-aea/blob/feature/protolint/scripts/generate_all_protocols.py#L330-L332

**Comment by marcofavorito** [comment]
Also, maybe some relevant parts of the docs should be updated to add protolint as necessary dependency.

File: .github/workflows/workflow.yml (hunk 2 above)
Comment (DavidMinarsch) [reply .github/workflows/workflow.yml:176]: I agree!

**Review by DavidMinarsch — COMMENTED** [review]
Screenshot 2021-01-25 at 13 23 02
some coverage gaps remain ^

File: aea/protocols/generator/common.py (hunk 4 above)
Comment (solarw) [reply aea/protocols/generator/common.py:367]: the reason, that windows generation differs from unix one cause new lines. so decided to unify it

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**